### PR TITLE
Added examples for hyper.appellf1 and Znm

### LINKED ISSUE #17887: Add missing docstring sections in Special submodule
As part of my Google Season of Docs project with SymPy, I copy edited all of the docstrings in the Special submodule to reflect the new SymPy Docstring Style Guide. While editing, I came across some docstrings that were missing sections or had other issues that need to be addressed. Here is a list of all of the functions in the Special submodule that had an issue and what needs to be fixed:

1. Jn_zeros 
Parameters listed but does not have a parameters section

2. Bspline_basis, bspline_basis_set
In bspline_basis, parameters not pulling into HTML site
Both have parameters listed but do not have a parameters section

3. Interpolating_spline
Parameters listed but does not have a parameters section

4. Fdiff, eval (under DiracDelta)
Parameters listed but do not have a parameters section

5. Fdiff, eval (under Heaviside)
Parameters listed but do not have a parameters section

6. E1(z)
Missing examples section

7. Li (in error functions)
In the Examples section, the example for under the text “Defining the ``li`` function via an integral:” is missing

8. Hyper
Lists properties as part of this class that need to be their own docstrings with their own examples

9. Meijerg
Lists properties as part of this class that need to be their own docstrings with their own examples

10. Appellf1
Missing examples section

11. Jacobi_normalized
Parameters listed but does not have a parameters section

12. Ynm_c
Missing examples section
Parameters listed but does not have a parameters section

13. Znm
Missing examples section

CC @sympy/gsod

### PR DESCRIPTION
**Pull request by Soumi7**
#### References to other Issues or PRs

Fixes #17887

#19591

#### Brief description of what is fixed or changed
Updated Functions.special submodule :
- Added examples for hyper.appellf1 in hyper.py
- Added examples for Znm in spherical_harmonics.py

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/functions/special/hyper.py [hunk 1]
```diff
@@ -1100,6 +1100,16 @@ class appellf1(Function):
         \frac{(a)_{m+n} (b_1)_m (b_2)_n}{(c)_{m+n}}
         \frac{x^m y^n}{m! n!}.
 
+    Examples
+    ========
+    >>> from sympy.functions.special.hyper import appellf1
```
Comment (Smit-create) [inline sympy/functions/special/hyper.py:1105]:
```suggestion
    ========

    >>> from sympy.functions.special.hyper import appellf1
```

File: sympy/functions/special/hyper.py [hunk 2]
```diff
@@ -1100,6 +1100,16 @@ class appellf1(Function):
...
+    >>> appellf1(2., 1., 6., 4., 5., 6.)
+    0.0063339426292673
+    >>> appellf1(1,2,3,5,0.5,0.25)
+    appellf1(1, 2, 3, 5, 0.5, 0.25)
```
Comment (Smit-create) [inline sympy/functions/special/hyper.py:1109]:
```suggestion
    >>> appellf1(1, 2, 3, 5, 0.5, 0.25)
    appellf1(1, 2, 3, 5, 0.5, 0.25)
```

File: sympy/functions/special/hyper.py [hunk 3]
```diff
@@ -1100,6 +1100,16 @@ class appellf1(Function):
...
+    >>> appellf1(1,2,3,5,0.5,0.25)
+    appellf1(1, 2, 3, 5, 0.5, 0.25)
```
Comment (Smit-create) [inline sympy/functions/special/hyper.py:1111]: +1

File: sympy/functions/special/spherical_harmonics.py [hunk 4]
```diff
@@ -262,6 +262,24 @@ def Ynm_c(n, m, theta, phi):
 
     Ynm, Znm
 
+    Examples
+    ========
+    >>> from sympy import Znm, Symbol, simplify
+    >>> from sympy.abc import n,m
```
Comment (Smit-create) [inline sympy/functions/special/spherical_harmonics.py:268]: Add a blank line after `======`
And spaces around commas.

File: sympy/functions/special/spherical_harmonics.py [hunk 5]
```diff
@@ -262,6 +262,24 @@ def Ynm_c(n, m, theta, phi):
...
+    >>> theta = Symbol("theta")
+    >>> phi = Symbol("phi")
+    >>> Znm(n, m, theta, phi)
+    Znm(n, m, theta, phi)
+
+    For specific integers n and m we can evaluate the harmonics
+    to more useful expressions:
+    >>> simplify(Znm(0, 0, theta, phi).expand(func=True))
```
Comment (Smit-create) [inline sympy/functions/special/spherical_harmonics.py:276]:
```suggestion
    to more useful expressions:

    >>> simplify(Znm(0, 0, theta, phi).expand(func=True))
```

File: sympy/functions/special/hyper.py (hunk 1 above)
Comment (Soumi7) [reply sympy/functions/special/hyper.py]: Done.

File: sympy/functions/special/hyper.py (hunk 3 above)
Comment (Soumi7) [reply sympy/functions/special/hyper.py]: Changed this.

File: sympy/functions/special/spherical_harmonics.py (hunk 4 above)
Comment (Soumi7) [reply sympy/functions/special/spherical_harmonics.py]: Done.

File: sympy/functions/special/spherical_harmonics.py (hunk 5 above)
Comment (Soumi7) [reply sympy/functions/special/spherical_harmonics.py]: Done.

File: sympy/functions/special/hyper.py (hunk 2 above)
Comment (Soumi7) [reply sympy/functions/special/hyper.py]: Changed.

File: sympy/functions/special/hyper.py [hunk 6]
```diff
@@ -1100,6 +1100,19 @@ class appellf1(Function):
...
         \frac{x^m y^n}{m! n!}.
 
+    Examples
+    ========
+
+    >>> from sympy.functions.special.hyper import appellf1
+    >>> appellf1(2., 1., 6., 4., 5., 6.)
+    0.0063339426292673
+    >>> appellf1(1, 2, 3, 5, 0.5, 0.25)
+    appellf1(1, 2, 3, 5, 0.5, 0.25)
```
Comment (sylee957) [inline sympy/functions/special/hyper.py:1110]: I don't think that the example to mix floats and rationals should be a model example.
We could probably have done to evaluate such cases numeric once there is there are any floats in the arguments.

Comment (Soumi7) [reply sympy/functions/special/hyper.py:1110]: Sure. Will change

Comment (asmeurer) [reply sympy/functions/special/hyper.py:1110]: At least one of the examples should show symbolic inputs.

Comment (asmeurer) [reply sympy/functions/special/hyper.py:1110]: Also if it is possible to find more instructive inputs than just 1, 2, 3, 4, 5, 6 that would help as well.

Comment (Soumi7) [reply sympy/functions/special/hyper.py]: Done!

**Pull request merged**